src/runtime-rs/crates/resource/src/share_fs/utils.rs

```diff
@@ -4,13 +4,27 @@
 // SPDX-License-Identifier: Apache-2.0
 //
 
-use std::path::{Path, PathBuf};
+use std::{
+    os::unix::fs::PermissionsExt,
+    path::{Path, PathBuf},
+};
 
 use anyhow::Result;
 use kata_sys_util::mount;
 
 use super::*;
 
+pub(crate) fn mkdir_with_permissions(path_target: PathBuf, mode: u32) -> Result<()> {
```

Comment:
I think you could replace [this](https://github.com/kata-containers/kata-containers/blob/main/src/runtime-rs/crates/resource/src/share_fs/virtio_fs_share_mount.rs#L82) with this function.

Reply:
What's about I replace {create_dir_all()...; set_permissions()} with the mkdir_with_permissions?

Reply:
Thx, fixed